File: backend/src/main/kotlin/network/bobnet/cms/model/content/Category.kt
package network.bobnet.cms.model.content

import network.bobnet.cms.model.data.RenderCategory
import java.time.LocalDateTime
import javax.persistence.*

@Entity
data class Category(
        var name: String = "",
        var description: String = "",
        var featuredImage: String = "",
        @Column(unique = true) var slug: String = "",
        var addedAt: LocalDateTime = LocalDateTime.now(),
        @Id @GeneratedValue var id: Long? = null,
        @ManyToOne var parent: Category? = null,
        @ManyToMany val articles: MutableSet<Article>? = null) {

    fun render() = RenderCategory(
            slug,
            name,
            featuredImage,
            description,
            parent,
            id
    )
}

File: backend/src/entity/content/category.ts
// @ts-ignore
import { Schema, Model } from "mongoose";
import {Long} from "long";

module.exports = Model('Category', new Schema({
    name: String,
    description: String,
    featuredImage: String,
    slug: String,
    addedAt: Date,
    id: Long,
    parent: Object,
    articles: Object
}));